### PR TITLE
Kind provisioner script fixup

### PR DESCRIPTION
**Pull request by su225**
https://github.com/istio/istio/pull/27248

After experimenting a bit, there were a few more fixes. As common-files is the golden source for `kind_provisioner.sh`, this has to be kept in sync.

### CODE REVIEW

**Review by nmittler — APPROVED** [review]

File: files/common/scripts/kind_provisioner.sh [hunk 1]
```diff
@@ -98,42 +98,54 @@ function cleanup_kind_cluster() {
...
+  if [[ -z "${DEFAULT_CLUSTER_YAML}" ]]; then
+    echo 'DEFAULT_CLUSTER_YAML file must be specified. Exiting...'
+    return 1
+  fi
+}
+
 # setup_kind_cluster creates new KinD cluster with given name, image and configuration
 # Even in case of single cluster, don't call this directly. Call through setup_kind_clusters
 # with cluster topology configuration file.
 # 1. CLUSTER_CONFIG: KinD cluster configuration YAML file (mandatory)
```
Comment (ericvn) [inline files/common/scripts/kind_provisioner.sh:111]: CLUSTER_CONFIG also removed

File: files/common/scripts/kind_provisioner.sh [hunk 2]
```diff
@@ -191,7 +200,7 @@ function setup_kind_clusters() {
     CLUSTER_YAML="${ARTIFACTS}/config-${CLUSTER_NAME}.yaml"
     if [ ! -f "${CLUSTER_YAML}" ]; then
       cp "${DEFAULT_CLUSTER_YAML}" "${CLUSTER_YAML}"
-      cat <<EOF >> "${CLUSTER_YAML}"
+        cat <<EOF >> "${CLUSTER_YAML}"
```
Comment (ericvn) [inline files/common/scripts/kind_provisioner.sh:203]: Line is indented two spaces too many.

Comment (su225) [reply files/common/scripts/kind_provisioner.sh:203]: Raised https://github.com/istio/common-files/pull/314

File: files/common/scripts/kind_provisioner.sh (hunk 1 above)
Comment (su225) [reply files/common/scripts/kind_provisioner.sh:111]: Raised https://github.com/istio/common-files/pull/314

**Pull request merged**